modules/cucumber-reportportal-formatter.js

```diff
@@ -346,41 +365,52 @@ const createRPFormatterClass = (config) => {
       ) {
         switch (event.media.type) {
           case 'text/plain': {
-            const logMessage = utils.getJSON(event.data);
+            const logObj = utils.getJSON(event.data);
             const request = {
               time: this.reportportal.helpers.now(),
             };
-            if (logMessage) {
-              request.level = logMessage.level;
-              request.message = logMessage.message;
+            let itemId = this.contextState.context.stepId;
+            if (logObj) {
+              request.level = logObj.level;
+              request.message = logObj.message;
+              if (logObj.entity === RP_ENTITY_LAUNCH) {
+                itemId = this.contextState.context.launchId;
+              }
             } else {
-              request.level = 'DEBUG';
+              request.level = LOG_LEVELS.DEBUG;
               request.message = event.data;
             }
-            this.reportportal.sendLog(this.contextState.context.stepId, request);
+            this.reportportal.sendLog(itemId, request);
             break;
           }
           default: {
+            const logObj = utils.getJSON(event.data);
             const request = {
               time: this.reportportal.helpers.now(),
-              level: this.contextState.context.stepStatus === STATUSES.PASSED ? 'DEBUG' : 'ERROR',
+              level:
+                this.contextState.context.stepStatus === STATUSES.PASSED
+                  ? LOG_LEVELS.DEBUG
+                  : LOG_LEVELS.DEBUG,
```

Comment:
Shouldn't these variables be different?

Reply:
Yeah! You are right. Thank you, it's a typo.